src/web/View.tsx

```diff
@@ -82,19 +82,17 @@ export class View extends ViewBase<Types.ViewProps, {}> {
         focusManager: PropTypes.object
     };
 
-    private focusManager: FocusManager;
+    private _focusManager: FocusManager;
 
-    private resizeDetectorAnimationFrame: number;
-    private resizeDetectorNodes: { grow?: HTMLElement, shrink?: HTMLElement } = {};
+    private _resizeDetectorAnimationFrame: number;
+    private _resizeDetectorNodes: { grow?: HTMLElement, shrink?: HTMLElement } = {};
 
     constructor(props: Types.ViewProps) {
         super(props);
-        if (this.props.restrictFocusWithin) {
-            this.focusManager = new FocusManager();
-        }
+        this._focusManager = new FocusManager();
```

Comment:
The problem is that getChildContext() which provides the children with appropriate FocusManager instance is invoked before the component is mounted (so, there would be no possibility to maintain the proper relations between instances). And React doesn't have a public API to traverse the child component instances (props.children provides properties and some other things, but not the exact instances and even no state).

What we can do is to force the `restrictFocusWithin` property to be readonly (and tell about it in the documentation) — it's sort of ugly, but makes sense, I think. I could go back to the initial implementation, but log an error in case restrictFocusWithin property is being changed during the component life cycle.